Add flowchart to explains the difference of `bootstrap_project` and `configure_project` (#3470)
kedro-org/kedro: docs/source/kedro_project_setup/session.md

- `save_on_close`: A boolean value to indicate whether or not to save the session to disk when it's closed
- `env`: Environment for the `KedroContext`
- `extra_params`: Optional dictionary containing extra project parameters
for the underlying **`KedroContext`**; if specified, this will update (and therefore take precedence over) parameters retrieved from the project configuration

## `bootstrap_project` and `configure_project`

> ```{image} ../meta/images/kedro-session-creation.png

Reviewer: Is there a reason for using this style of image rather than the `! []()` markdown notation? Reason I ask is that it doesn't render on GitHub and I think it would help our readers to have visibility on the graphic here if possible (and you can add `alt` tags if you want them, so no need to use the `rst` image syntax -- AFAICT?).

Author: very good question! I don't have a good idea, it appears to me @astrojuanlu did something to all mermaid diagram with this new style. I also found it annoying it won't render in VSCode.

Reviewer 2: Block-level images aren't really needed https://myst-parser.readthedocs.io/en/latest/syntax/images_and_figures.html#block-level-images I kept them for consistency, they can be turned into inline images

:alt: mermaid-General overview diagram for KedroSession creation
```

% Mermaid code, see https://github.com/kedro-org/kedro/wiki/Render-Mermaid-diagrams
% graph LR
%  subgraph Kedro Startup Flowchart
%    A[bootstrap_project] -->|Read pyproject.toml| B
%    A -->|Add project root to sys.path| B[configure_project]
%    C[Initialize KedroSession]
%    B --> |Read settings.py| C
%    B --> |Read pipeline_registry.py| C
%  end

> Both `bootstrap_project` and `configure_project` handle the setup of a Kedro project, but there are subtle differences. `bootstrap_project` is used for project mode, and `configure_project` is used for packaged mode.

Reviewer: ```suggestion
Both `bootstrap_project` and `configure_project` handle the setup of a Kedro project, but there are subtle differences: `bootstrap_project` is used for project mode, and `configure_project` is used for packaged mode.
```

> In most cases, you don't need to use these functions because Kedro's CLI already runs them at startup, i.e. `kedro run`. If you want to [interact with Kedro Project programtically in an interactive session such as Notebook](../notebooks_and_ipython/kedro_and_notebooks.md#reload_kedro-line-magic), you can use the `%reload_kedro` with Jupyter or IPython. You should only use these functions directly if none of these methods work.

Reviewer: ```suggestion
Kedro's CLI runs the functions at startup as part of `kedro run` so in most cases you don't need to call these functions. If you want to [interact with a Kedro project programatically in an interactive session such as Notebook](../notebooks_and_ipython/kedro_and_notebooks.md#reload_kedro-line-magic), use `%reload_kedro` line magic with Jupyter or IPython. Only use these functions directly if none of these methods work.
```

Author: Thanks! I like the edited version more!

### `bootstrap_project`

> This function uses `configure_project` under the hood. Additionally, it reads metadata from `pyproject.toml` and adds the project root to `sys.path`. This ensures your project can be imported as a Python package. This is usually used when you are working with the source code of a Kedro Project.

Reviewer: ```suggestion
This function uses `configure_project` under the hood, and additionally reads metadata from `pyproject.toml` and adds the project root to `sys.path` so the project can be imported as a Python package. It is typically used to work directly with the source code of a Kedro project.
```

### `configure_project`

> This function reads `settings.py` and `pipeline_registry.py` and registers the configuration before Kedro's run starts. If you have an packaged Kedro project, you only need to run `configure_project` before executing your pipeline.

Reviewer 3: ```suggestion
This function reads `settings.py` and `pipeline_registry.py` and registers the configuration before Kedro's run starts. If you have a packaged Kedro project, you only need to run `configure_project` before executing your pipeline.
```
